CI.Jenkinsfile

```diff
@@ -0,0 +1,52 @@
+
+
+def versions = [3.11, 3.10, 3.9, 3.8, 3.7]
+
+def runSonnarForPythonVersion(sourceDir, ver){
+    mySonarOpts="-Dsonar.sources=/source -Dsonar.host.url=${env.SONAR_HOST_URL} -Dsonar.login=${env.SONAR_AUTH_TOKEN}"
+    if("${env.CHANGE_ID}" != "null"){
+        mySonarOpts = "$mySonarOpts -Dsonar.pullrequest.key=${env.CHANGE_ID} -Dsonar.pullrequest.branch=${env.BRANCH_NAME}"
+    } else {
+        mySonarOpts = "$mySonarOpts -Dsonar.branch.name=${env.BRANCH_NAME}"
+    } 
+    if ("${env.CHANGE_BRANCH}" != "null") {
+        mySonarOpts="$mySonarOpts -Dsonar.pullrequest.base=${env.CHANGE_TARGET} -Dsonar.pullrequest.branch=${env.CHANGE_BRANCH}"
+    }
+    sh "docker run \
+            --pull always \
+            --rm --volume ${sourceDir}:/source \
+            python:${ver}-slim \
+            bash -c \"apt-get update && \
+            apt-get install -y wget unzip && \
+            pip3 install tox && \
+            cd /root/ && \
+            wget -q https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.8.0.2856-linux.zip && \
```

Comment:
Leave a TODO comment somewhere in here to pull out the Sonar run to a single Python version (the latest?).  We only need to download the archive and execute it once.  Not for each Python version.

Reply:
TODO comment added!